rotkehlchen/chain/gnosis/modules/sdai/decoder.py

```diff
@@ -0,0 +1,114 @@
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.constants import \
+    DEFAULT_TOKEN_DECIMALS  # all tokens in this decoder use default(18)
+from rotkehlchen.chain.evm.decoding.constants import (
+    CPT_SDAI,
+    SDAI_CPT_DETAILS,
+    SDAI_DEPOSIT,
+    SDAI_REDEEM,
+)
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.gnosis.modules.sdai.constants import GNOSIS_SDAI_ADDRESS
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.gnosis.node_inquirer import GnosisInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+class SdaiDecoder(DecoderInterface):
+    def __init__(
+            self,
+            gnosis_inquirer: 'GnosisInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=gnosis_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+
+    def _decode_sdai_deposit_events(self, context: DecoderContext) -> DecodingOutput:
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = token_normalized_value_decimals(amount_raw, DEFAULT_TOKEN_DECIMALS)
+        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:64])
+        shares = token_normalized_value_decimals(shares_raw, DEFAULT_TOKEN_DECIMALS)
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount
+            ):
+                asset = event.asset.resolve_to_crypto_asset()
+                event.counterparty = CPT_SDAI
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {asset.symbol} into the savings xDAI contract'  # noqa: E501
```

Comment:
I told you to capitalize this in discord

```suggestion
                event.notes = f'Deposit {event.balance.amount} {asset.symbol} into the Savings xDAI contract'  # noqa: E501
```